UnityProject/Assets/Scripts/Items/Cargo/MiningScanner.cs

```diff
@@ -0,0 +1,52 @@
+using System.Collections;
+using Core.Lighting;
+using Mirror;
+using UnityEngine;
+
+namespace Items.Cargo
+{
+	public class MiningScanner : NetworkBehaviour, IInteractable<HandActivate>
+	{
+
+		public bool IsAdvanced = false;
+		public float NormalScanCooldown = 10f;
+		public float AdvancedScanCooldown = 4f;
+
+		private bool isOnCooldown = false;
+
+
+		[TargetRpc]
+		public void TargetStartScanning(NetworkConnection target)
+		{
+			Debug.Log("scanning");
+			HighlightScanManager.Instance.Highlight();
+		}
+
+		public void ServerPerformInteraction(HandActivate interaction)
+		{
+			if (isOnCooldown)
+			{
+				Chat.AddExamineMsg(interaction.Performer, $"The {gameObject.ExpensiveName()} is still recharging!");
+				return;
+			}
+
+			if (CustomNetworkManager.IsHeadless)
+			{
+				TargetStartScanning(interaction.PerformerPlayerScript.connectionToClient);
+			}
+			else
+			{
+				HighlightScanManager.Instance.Highlight();
+			}
```

Comment:
Does just calling the RPC not work in the editor?

Reply:
it doesn't